src/main/java/htsjdk/samtools/cram/io/CramInt.java

```diff
@@ -39,8 +39,11 @@ public static int int32(final byte[] data) {
      * @param buffer {@link ByteBuffer} to read from
      * @return an integer value read from the buffer
      */
-    public static int int32(final ByteBuffer buffer) {
-        return buffer.get() | buffer.get() << 8 | buffer.get() << 16 | buffer.get() << 24;
+    public static int readInt32(final ByteBuffer buffer) {
+        return (0xFF & (int) buffer.get()) |
+                (0xFF & (int) buffer.get()) << 8 |
```

Comment:
casting to `int` should be unnecessary here

Reply:
I'll remove the int casts - good catch.

`0xFF &` is still necessary because int-promotion extends the negative sign into higher bytes.

Reply:
OK I now see why this is necessary, I forgot that byte isn't an unsigned type. Why, Java? Why?